language/stdlib/modules/DiemTransactionPublishingOption.move

```diff
@@ -139,6 +127,44 @@ module DiemTransactionPublishingOption {
         include DiemConfig::SetAbortsIf<DiemTransactionPublishingOption>{account: dr_account};
     }
 
+    /// If published, halts transactions from all accounts except DiemRoot
+    resource struct HaltAllTransactions {}
+
+    /// If called, transactions cannot be sent from any account except DiemRoot
+    public fun halt_all_transactions(dr_account: &signer) {
+        Roles::assert_diem_root(dr_account);
+        assert(
+            !exists<HaltAllTransactions>(Signer::address_of(dr_account)),
+            Errors::already_published(EHALT_ALL_TRANSACTIONS),
+        );
+        move_to(dr_account, HaltAllTransactions {});
+
+        // trigger reconfig
+        let publish_option = DiemConfig::get<DiemTransactionPublishingOption>();
+        DiemConfig::set<DiemTransactionPublishingOption>(dr_account, publish_option);
```

Comment:
This and in the function below don't seem to change anything, other than they are being used to trigger a reconfiguration. What would you think about just directly triggering the reconfiguration here and below (through `DiemConfig::reconfigure`) instead?

Reply:
I didn't realize that was public (and am slightly disturbed :)), but I agree that is much cleaner. Will fix.

Reply:
Actually, now that you mention it: I don't think we need to trigger reconfiguration at all here. When the mechanism for halting txes was the allowlist, that needed reconfig because the VM reads it from on-chain + caches it. Now, the mechanism is a simple data change and will take effect immediately.

Reply:
Even better!